### DISCUSSION TITLE
technical questions for dev team

### DISCUSSION THREAD

**Discussion by DMaro28** [microsoft/typespec#2374] [Q&A]
My company is using Typespec extensively and planning to further extend the usage to manage versioning. Some technical questions came up for the dev team:

- Are there limits on decorator numbers in a project? We expect tens of thousand in active usage.
- Is there an option to generate only the latest version (or couple of versions) of the openapi file when using versioning module?
- What's the expected timeline to attain stable version 1.0 of TypeSpec?

**Comment by bterlson** [comment]
This is exciting to hear!

There are no limits to the number of decorators. That said, we have not focused much on performance, so you may find issues at a large enough scale. Such issues would be extremely valuable to file if you find them!

There is no option to generate a subset of versions or a specific version, but thats a good feature request. Can you file an issue?

I don't know when 1.0 stable will release, but we are working toward a 1.0 beta release in the near future. Much will depend on the feedback we receive, including yours! We are using TypeSpec extensively at Microsoft so are quite motivated to avoid any major breaking changes and I would expect TypeSpec to be fairly stable even at its current release.

Would.love to hear more about what you're building! If you'd be interested in meeting the team for show and tell let me know 😁

**Reply by DMaro28** [reply]
Thanks Brian for the prompt response, much appreciated! I will create the issue requesting a subset of version from versioning module. In the spirit of introductions, I work for LSEG, we are building a rather large set of APIs for Microsoft/LSEG partnership using Typespec. We've already had an introductory meeting with your product team. I would be great to meet your team as well!

**Reply by DMaro28** [reply]
@bterlson  I tried to reach out via email. I'm not sure I got through. Are you still open to arranging a meeting between our teams?
